File: sorting/bubblesort/bubblesort.js
const DATA = [1, 4, 6, 8, 3, 9, 2, 7, 5]

/**
 * sort
 *
 * @param   {Array} data
 * @returns {Array}
 */
const sort = data => {
  let n = data.length,
    swapped = true

  do {
    swapped = false
    for (let i = 0; i < n - 1; i++) {
      if (data[i] > data[i + 1]) {
        let temp = data[i]
        data[i] = data[i + 1]
        data[i + 1] = temp

        swapped = true
      }
    }
    n = n - 1
  } while (swapped)

  return data
}

console.log(sort(DATA))

File: sorting/bubblesort/bubblesort.kt
package sorting.bubblesort

fun main(args: Array<String>) {
  val data: IntArray = intArrayOf(1, 4, 6, 8, 3, 9, 2, 7, 5)
  Sort(data)
}

fun Sort(data: IntArray) {
  var n = data.size
  var swapped = true

  while (swapped) {
    swapped = false

    var i = 0
    while (i < n - 1) {
      if (data[i] > data[i + 1]) {
        var temp = data[i]
        data[i] = data[i + 1]
        data[i + 1] = temp
        swapped = true
      }
      i++
    }
    n = n - 1
  }

  for (value in data) {
    println(value)
  }
}
